accept.go

```diff
@@ -150,12 +152,34 @@ func accept(w http.ResponseWriter, r *http.Request, opts *AcceptOptions) (_ *Con
 	b, _ := brw.Reader.Peek(brw.Reader.Buffered())
 	brw.Reader.Reset(io.MultiReader(bytes.NewReader(b), netConn))
 
+	ipStr, portStr, err := net.SplitHostPort(r.RemoteAddr)
+	if err != nil {
+		err = errors.New("*http.Request contains invalid RemoteAddr")
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
```

Comment:
Note: Since the connection has been hijacked, we can't use `http.Error` or `w` here. If we'd want to tell the client about this issue, it should happen before switching protocols and hijacking.

(That said, this new behavior shouldn't return any errors here, see previous notes.)

Reply:
got it.